mm2src/coins/eth.rs

```diff
@@ -6130,52 +6374,151 @@ impl From<Web3RpcError> for EthGasDetailsErr {
     fn from(e: Web3RpcError) -> Self {
         match e {
             Web3RpcError::Transport(tr) | Web3RpcError::InvalidResponse(tr) => EthGasDetailsErr::Transport(tr),
-            Web3RpcError::Internal(internal) | Web3RpcError::Timeout(internal) => EthGasDetailsErr::Internal(internal),
+            Web3RpcError::Internal(internal)
+            | Web3RpcError::Timeout(internal)
+            | Web3RpcError::InvalidGasApiConfig(internal) => EthGasDetailsErr::Internal(internal),
             Web3RpcError::NftProtocolNotSupported => EthGasDetailsErr::NftProtocolNotSupported,
         }
     }
 }
 
-async fn get_eth_gas_details(
+async fn get_eth_gas_details_from_withdraw_fee(
     eth_coin: &EthCoin,
     fee: Option<WithdrawFee>,
     eth_value: U256,
     data: Bytes,
     call_addr: Address,
     fungible_max: bool,
 ) -> MmResult<GasDetails, EthGasDetailsErr> {
-    match fee {
+    let pay_for_gas_option = match fee {
         Some(WithdrawFee::EthGas { gas_price, gas }) => {
-            let gas_price = wei_from_big_decimal(&gas_price, 9)?;
-            Ok((gas.into(), gas_price))
+            let gas_price = wei_from_big_decimal(&gas_price, ETH_GWEI_DECIMALS)?;
+            return Ok((gas.into(), PayForGasOption::Legacy(LegacyGasPrice { gas_price })));
+        },
+        Some(WithdrawFee::EthGasEip1559 {
+            max_fee_per_gas,
+            max_priority_fee_per_gas,
+            gas_option: gas_limit,
+        }) => {
+            let max_fee_per_gas = wei_from_big_decimal(&max_fee_per_gas, ETH_GWEI_DECIMALS)?;
+            let max_priority_fee_per_gas = wei_from_big_decimal(&max_priority_fee_per_gas, ETH_GWEI_DECIMALS)?;
+            if let EthGasLimitOption::Set(gas) = gas_limit {
+                return Ok((
+                    gas.into(),
+                    PayForGasOption::Eip1559(Eip1559FeePerGas {
+                        max_fee_per_gas,
+                        max_priority_fee_per_gas,
+                    }),
+                ));
+            } else {
+                // go to gas estimate code
+                PayForGasOption::Eip1559(Eip1559FeePerGas {
+                    max_fee_per_gas,
+                    max_priority_fee_per_gas,
+                })
```

Comment:
Match should be used here instead of if else
```rust
            match gas_limit {
                EthGasLimitOption::Set(gas) => {
                    return Ok((
                        gas.into(),
                        PayForGasOption::Eip1559(Eip1559FeePerGas {
                            max_fee_per_gas,
                            max_priority_fee_per_gas,
                        }),
                    ));
                },
                EthGasLimitOption::Calc => {
                    // go to gas estimate code
                    PayForGasOption::Eip1559(Eip1559FeePerGas {
                        max_fee_per_gas,
                        max_priority_fee_per_gas,
                    })
                },
            }
```

Reply:
done in [531f1af](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2051/commits/531f1afb6832974583ae7110575769b952393e77)